da/celestia/celestia.go

```diff
@@ -223,36 +232,105 @@ func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 				continue
 			}
 
-			//double check txResponse is not nil - not supposed to happen
-			if txResponse == nil {
-				err := errors.New("txResponse is nil")
-				c.logger.Error("Failed to submit DA batch", "error", err)
+			daMetaData := &da.DAMetaData{
+				Height:      height,
+				Commitments: commitments,
+			}
+
+			availabilityResult := c.CheckBatchAvailability(daMetaData)
+			if availabilityResult.Code != da.StatusSuccess || !availabilityResult.DataAvailable {
+				c.logger.Error("Failed to submit DA batch. Blob not available")
 				res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, false, err)
 				if err != nil {
 					return res
 				}
 				time.Sleep(c.submitRetryDelay)
 				continue
 			}
-
-			c.logger.Info("Successfully submitted DA batch", "txHash", txResponse.TxHash, "daHeight", txResponse.Height, "gasWanted", txResponse.GasWanted, "gasUsed", txResponse.GasUsed)
 			res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, true, nil)
 			if err != nil {
 				return res
 			}
 			return da.ResultSubmitBatch{
 				BaseResult: da.BaseResult{
 					Code:     da.StatusSuccess,
-					Message:  "tx hash: " + txResponse.TxHash,
-					DAHeight: uint64(txResponse.Height),
+					Message:  "Submission successful",
+					MetaData: availabilityResult.MetaData,
+				},
+			}
+		}
+	}
+
+}
+
+func (c *DataAvailabilityLayerClient) RetrieveBatches(daMetaData *da.DAMetaData) da.ResultRetrieveBatch {
+
+	//Just for backward compatibility, in case no commitments are sent from the Hub, batch can be retrieved using previous implementation.
+	if daMetaData.Commitments == nil || len(daMetaData.Commitments) == 0 {
```

Comment:
This is actually a security hole so not sure we can release it like that.
The big majority of Froopyland don't run full nodes. Once we upgrade them assuming not synced, they will have to sync from a snapshot from the point that state updates must contain commitments.
I suggest removing it.